azure/virtual-machine/examples/domain-joined/main.tf

```diff
@@ -0,0 +1,79 @@
+provider "azurerm" {
+  features {
+    resource_group {
+      prevent_deletion_if_contains_resources = false
+    }
+    key_vault {
+      purge_soft_delete_on_destroy = true
+    }
+  }
+}
+
+locals {
+  location = "uksouth"
+  tags = {
+    module  = "virtual-machine"
+    example = "domain-joined"
+    usage   = "demo"
+  }
+  resource_prefix = "vm-dj-demo-uks-03"
+}
+
+resource "azurerm_resource_group" "vm" {
+  name     = "rg-${local.resource_prefix}"
+  location = local.location
+  tags     = local.tags
+}
+
+resource "azurerm_virtual_network" "vm" {
+  name                = "vnet-${local.resource_prefix}"
+  resource_group_name = azurerm_resource_group.vm.name
+  location            = local.location
+  tags                = local.tags
+
+  address_space = ["10.0.0.0/24"]
+  dns_servers   = ["10.0.0.10", "10.0.0.11"]
+}
+
+resource "azurerm_subnet" "vm" {
+  name                 = "snet-vm"
+  resource_group_name  = azurerm_resource_group.vm.name
+  virtual_network_name = azurerm_virtual_network.vm.name
+
+  address_prefixes = azurerm_virtual_network.vm.address_space
+}
+
+module "vm" {
+  source = "../../"
+
+  name                = "vm-${local.resource_prefix}"
+  location            = local.location
+  resource_group_name = azurerm_resource_group.vm.name
+  tags                = local.tags
+
+  os_type       = "Windows"
+  computer_name = "vm"
+  password      = "P4s5w0rd!"
```

Comment:
I feel that it would make more sense to put this "password" into a local. That way if someone copied and pastes this portion of the code, they're not copying this terrible password 😄

Reply:
Going to add a sensitive var to handle this instead